metrics/mixin/README.md

```diff
@@ -0,0 +1,76 @@
+# Prometheus Monitoring Mixin for OCS
+
+A set of Prometheus alerts for OCS.
+
+The scope of this directory is to provide OCS specific Prometheus rule files using Prometheus Mixins.
+
+## Prerequisites
+* Jsonnet [[Install Jsonnet]](https://github.com/google/jsonnet#building-jsonnet)
+
+   [Jsonnet](https://jsonnet.org/learning/getting_started.html) is a data templating language for app and tool developers.
+
+   The mixin project uses Jsonnet to provide reusable and configurable configs for Grafana Dashboards and Prometheus Alerts.
+* Jsonnet-bundler [[Install Jsonnet-bundler]](https://github.com/jsonnet-bundler/jsonnet-bundler#install)
+
+   [Jsonnet-bundler](https://github.com/jsonnet-bundler/jsonnet-bundler) is a package manager for jsonnet.
+* Promtool
+  1. [Download](https://golang.org/dl/) Go (>=1.11) and [install](https://golang.org/doc/install) it on your system.
+  2. Setup the [GOPATH](http://www.g33knotes.org/2014/07/60-second-count-down-to-go.html) environment.
+  3. Run `$ go get -d github.com/prometheus/prometheus/cmd/promtool`  
+
+
+## How to use?
+
+**To get dependencies**
+
+`$ jb install` inside the directory **metrics/mixin/build**
+
+### To add new alerts/rules:
+
+* Add new alerts in 'metrics/mixin/alerts' directory in libsonnet format. Learn about Jsonnet and Libsonnet from https://jsonnet.org/.
```

Comment:
This is not sufficient to even explain your example. For instance, I have no idea what `+::` does, as it is mentioned nowhere in the Jsonnet website. You can't assume that anyone coming to this repo has any idea how to work with jsonnet, or even prometheus mixins. The design doc is a bit too high-level and verbose to tell a developer how to modify this structure.

We're also missing and explanation of what the various files in this directory are for. For instance, why are there both `services-external.libsonnet` and `services.libsonnet` when they appear to be identical?

Reply:
@jarrpa Its a new language and its not just us using it, It's a standard in Kubernetes monitoring world. Eg https://github.com/openshift/cluster-monitoring-operator/, https://github.com/kubernetes-monitoring/kubernetes-mixin. Just like I had to learn Golang to contribute to the OCS-operator, Jsonnet is a requirement to learn here. Also in OCS operator we have two active contributors for this, me and @umangachapagain and will be mostly. If more contributors have to contribute, they will have to learn.

Regarding the documentation, it's not as simple to get it, one has to do the whole tutorial to learn. Me and @umangachapagain have been through the pain. 

Regarding keeping two different files, I will give you an example: https://github.com/rook/rook/tree/master/cluster/examples/kubernetes/ceph/monitoring. If you have a look at the files in this repo you will see that external mode files have way lesser alerts than internal as they are no applicable there. 
We will have the same situation in OCS-operator as well, as the list of the alerts grows. Also, the OCS-operator alerts are not identical, there is a slight difference in the message and description fields which is because of the difference between the implementation of the internal and external mode of cephobjectstore.

Reply:
This is worlds different than just knowing Golang. But even there, I have a todo item to improve the on-boarding documentation so people can get started with minimal working knowledge in Go. Best I can tell, Jsonnet doesn't have nearlyt he same market penetration as Go either, so you can't just assume people will be able to pick it up with the same ease. I don't want to skip that here, especially because *I* need to know what's going on before I can approve anything.

If you have the relevant Jsonnet tutorials, link those in the README. Any maintainer who is neither you or Umanga should be able to look at this PR and understand what is happening. Links to the projects you just referenced probably wouldn't hurt, either.